clang/include/clang/AST/ASTStructuralEquivalence.h

```diff
@@ -43,6 +43,9 @@ struct StructuralEquivalenceContext {
   /// key: (from, to, IgnoreTemplateParmDepth)
   using NonEquivalentDeclSet = llvm::DenseSet<std::tuple<Decl *, Decl *, int>>;
 
+  /// The language options to use for making a structural equivalence check.
+  const LangOptions &LangOpts;
+
```

Comment:
We already have 2 ASTContext; why not use the LangOpt from there? Possibly with a small function to make it less verbose.

Reply:
Good question! I originally did that, but the problem there is: those AST contexts can be different *and can have different language options set*. So I made it explicit and force the caller to set the language options.